tools/swift-format/Sources/Rules/BeginDocumentationCommentWithOneLineSummary.swift

```diff
@@ -7,6 +7,103 @@ import SwiftSyntax
 /// Lint: If a comment does not begin with a single-line summary, a lint error is raised.
 ///
 /// - SeeAlso: https://google.github.io/swift#single-sentence-summary
-public final class BeginDocumentationCommentWithOneLineSummary: SyntaxLintRule {
+public final class BeginDocumentationCommentWithOneLineSummary:  SyntaxLintRule {
+  override public func visit(_ node: FunctionDeclSyntax) {
+    diagnoseDocComments(node)
+  }
 
+  override public func visit(_ node: EnumDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: InitializerDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: DeinitializerDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: SubscriptDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: ClassDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: VariableDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: StructDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: ProtocolDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  override public func visit(_ node: TypealiasDeclSyntax) {
+    diagnoseDocComments(node)
+  }
+
+  /// Diagnose documentation comments that doesn't starts
+  /// with one sentence summary.
+  func diagnoseDocComments(_ decl: DeclSyntax) {
+    guard let commentText = decl.docComment else { return }
+    let docComments = commentText.components(separatedBy: "\n")
+    guard let firstPart = getFirstPartOfTheComment(docComments) else { return }
+
+    let sentences = getSentences(firstPart)
+    if sentences.count > 1 {
+      diagnose(.declRequiresBlankComment(sentences.first!), on: decl)
+    }
+  }
+
+  /// Returns the text of the first part of the comment,
+  func getFirstPartOfTheComment(_ comments: [String]) -> String? {
```

Comment:
Naming suggestion: `firstParagraph` instead of `getFirstPartOfTheComment`. (Functions in Swift traditionally do not start with "get", but rather are noun phrases that describe what the return value is.)